### DISCUSSION TITLE
Is a 4 minute build time normal for larger Nuxt v3 projects?

### DISCUSSION THREAD

**Discussion by timb-103** [nuxt/nuxt#20642] [Questions]
My project is pretty big:

- 350 API routes
- 400 components

The build time is around 4 minutes, which seems okay but I'm wondering if it should/could be faster?

Anyone else have a large project with similar times?

I should also mention, build time on the same project with Nuxt v2 is 2-3 minutes, Nuxt v3 is 4 minutes.

**Comment by warflash** [comment]
I'm getting similar times with projects of roughly that size I work on so seems reasonable I'd say

**Reply by timb-103** [reply]
Thanks for the insight!

**Comment by huang-julien** [comment]
Depending on what you're using, yes 4 minutes can be "okay". Having a smaller project but with webpack/bridge, it can take up to 5mn to build for me.

**Reply by timb-103** [reply]
The issue is, with Nuxt 2 the same project takes 2-3mins but then Nuxt 3 it's 4 mins. I would of assumed it would be faster lol.

**Reply by huang-julien** [reply]
I wish i could have only 2-3 minutes of build on webpack :joy: . Some perf improvement are coming in 3.5

**Comment by matthew-dean** [comment]
I'm not surprised at this build times (well... somewhat surprised -- 4 minutes is a bit much). But why is this number a constant? Why aren't there incremental rebuilds? When very few files have actually been changed, then a build should take milliseconds, in theory.

Also, why does Nuxt traverse the entire module graph up front, instead of incrementally building per the dependency tree of each navigated route when in dev mode?
